src/blockchain/mod.rs

```diff
@@ -164,6 +164,65 @@ pub trait ConfigurableBlockchain: Blockchain + Sized {
     fn from_config(config: &Self::Config) -> Result<Self, Error>;
 }
 
+/// Trait for blockchains that don't contain any state
+///
+/// Statless blockchains can be used to sync multiple wallets with different descriptors.
+///
+/// [`BlockchainFactory`] is automatically implemented for `Arc<T>` where `T` is a stateless
+/// blockchain.
+pub trait StatelessBlockchain: Blockchain {}
+
+/// Trait for a factory of blockchains that share the underlying connection or configuration
+#[cfg_attr(
+    not(feature = "async-interface"),
+    doc = r##"
+## Example
+
+This example shows how to sync multiple walles and return the sum of their balances
+
+```no_run
+# use bdk::Error;
+# use bdk::blockchain::*;
+# use bdk::database::*;
+# use bdk::wallet::*;
+fn sum_of_balances<B: BlockchainFactory>(blockchain_factory: B, wallets: &[Wallet<MemoryDatabase>]) -> Result<u64, Error> {
+    Ok(wallets
+        .iter()
+        .map(|w| -> Result<_, Error> {
+            w.sync(&blockchain_factory.build("wallet_1", None)?, SyncOptions::default())?;
```

Comment:
Would it be more clear to use the `wallet_name_from_descriptor()` function in the example?
```suggestion
            let wallet_name = wallet_name_from_descriptor(...);
            w.sync(&blockchain_factory.build(wallet_name, None)?, SyncOptions::default())?;
```

Reply:
I could use the descriptor checksum, just because it's shorter and easier to fit in a doc example